### PR TITLE
Tests AppModules 

### PR DESCRIPTION
**Pull request by Sergey-snayp**
Tests for all functions AppModules

### CODE REVIEW

File: test/models/AppModel/test.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+describe('App Model', async () => {
+  describe('On getOne', async () => {
+    let app, result;
+    beforeEach(async () => {
+      app = await AppFactory.Create();
+    });
+    it('Should check names for identity', async () => {
+      result = await AppModel.getOne({ name: app.name });
+      expect(result.app).to.deep.eq(app._doc);
```
Comment (Waiviogit) [inline test/models/AppModel/test.js]: In this test you check whole apps for identity, not names. Rename title

File: test/models/AppModel/test.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    });
+    it('Should check that the error exists', async () => {
+      result = await AppModel.getOne({ name: faker.random.string() });
+      expect(result.error).is.exist;
```
Comment (Waiviogit) [inline test/models/AppModel/test.js]: expect(result.error).to.be.exist     (if you will completely write off the tests from the second project, I will write comments on each, write yourself)

File: test/models/AppModel/test.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    });
+    it('Should return error message', async () => {
+      result = await AppModel.getOne({ name: faker.random.string() });
+      expect(result.error.message).to.eq('App not found!');
```
Comment (Waiviogit) [inline test/models/AppModel/test.js]: expect(result.error.message).to.be.eq('App not found!');

File: test/models/AppModel/test.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        {
+          $project: {
+            _id: 0, name: 1, admins: 1,
+          },
+        },
+      ]);
+      [result] = aggResult.result;
+      expect(result).to.be.undefined;
+    });
+  });
```
Comment (Waiviogit) [inline test/models/AppModel/test.js]: in fact, you aggregate the same record in each test, and just use the project to change the display ... There should be one test to verify the project, aggregation is a very extensive method, you can check its other capabilities, read about aggregations, in aggregation, you can sort, group, and much more, you do not need to write a bunch of tests for the same operation, in fact they are useless

File: test/models/AppModel/test.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  describe('On updateOne', async () => {
+    let name, app1, app, result;
+    beforeEach(async () => {
+      name = faker.name.firstName();
+      app1 = await AppFactory.Create({ name });
+    });
+    it('Update name at our request and get true', async () => {
+      result = await AppModel.updateOne({ name, updData: { $set: { name: faker.name.firstName() } } });
+      expect(result.result).to.be.true;
```
Comment (Waiviogit) [inline test/models/AppModel/test.js]: Test case must start with Should, dont write 'our' in test cases

File: test/models/AppModel/test.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    });
+    it('Should return false, when the data has not updated', async () => {
+      result = await AppModel.updateOne({ name: 'null', updData: { $set: { name: faker.name.firstName() } } });
+      expect(result.result).to.be.false;
+    });
+    it('Should return duplicate key error, when the data has duplicate name', async () => {
+      app = await AppFactory.Create({ name: faker.name.firstName() });
+      result = await AppModel.updateOne({ name: app1.name, updData: { $set: { name: app.name } } });
+      expect(result.error).is.exist;
```
Comment (Waiviogit) [inline test/models/AppModel/test.js]: expect(result.error).to.be.exist; You have to write in one style.

File: test/models/AppModel/test.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
+const {
+  expect, AppModel, faker, dropDatabase,
+} = require('test/testHelper');
+const { AppFactory } = require('test/factories/');
```
Comment (olegtsigulov) [inline test/models/AppModel/test.js]: delete last / (after factories)

File: test/models/AppModel/test.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+const _ = require('lodash');
+
+describe('App Model', async () => {
+  describe('On getOne', async () => {
+    let app, result;
+    beforeEach(async () => {
```
Comment (olegtsigulov) [inline test/models/AppModel/test.js]: make result const in each test, you dont use it in beforeEach case, i told you

File: test/models/AppModel/test.js [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+      result = await AppModel.getOne({ name: faker.random.string() });
+      expect(result.error).to.be.exist;
+    });
+    it('Should return error message', async () => {
+      result = await AppModel.getOne({ name: faker.random.string() });
+      expect(result.error.message).to.be.eq('App not found!');
+    });
+  });
+  describe('On getAll', async () => {
+    let result, appsCount;
```
Comment (olegtsigulov) [inline test/models/AppModel/test.js]: make result const!

File: test/models/AppModel/test.js [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+        }
+      });
+      it('Should check that getAll returns correct number of records', async () => {
+        result = await AppModel.getAll();
+        expect(result.apps.length).to.be.eq(appsCount);
+      });
+    });
+    describe('On error case', async () => {
+      beforeEach(async () => {
+        result = await AppModel.getAll();
```
Comment (olegtsigulov) [inline test/models/AppModel/test.js]: let result; (Add this before  beforeEach hook)

File: test/models/AppModel/test.js [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+      });
+      it('Should check that the error exists', async () => {
+        expect(result.error).is.exist;
```
Comment (olegtsigulov) [inline test/models/AppModel/test.js]: to.be ??

File: test/models/AppModel/test.js [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+      });
+      it('Should return error message', async () => {
+        expect(result.error.message).to.be.eq('App not found!');
+      });
+    });
+  });
+  describe('On aggregate', async () => {
+    let name, admins, blacklists, app, aggResult, result;
```
Comment (olegtsigulov) [inline test/models/AppModel/test.js]: aggResult, result - you dont use it at beforeEach, make it const at each case!

File: test/models/AppModel/test.js [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+    beforeEach(async () => {
+      await dropDatabase();
+      name = faker.name.firstName();
+      admins = [faker.name.firstName().toLowerCase()];
+      blacklists = [faker.name.firstName().toLowerCase()];
+      app = await AppFactory.Create({
+        name, admins, blacklists,
+      });
+    });
+    it('Should return from our request: id, name and admins. Using group stage', async () => {
```
Comment (olegtsigulov) [inline test/models/AppModel/test.js]: dont write 'our'

File: test/models/AppModel/test.js [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+      aggResult = await AppModel.aggregate([
+        { $match: { name: app.name } },
+      ]);
+      [result] = aggResult.result;
+      expect(result).to.deep.eq(app._doc);
+    });
+    it('Should return undefined, when aggregate has not result', async () => {
+      aggResult = await AppModel.aggregate([
+        { $match: { surname: app.name } },
+        {
```
Comment (olegtsigulov) [inline test/models/AppModel/test.js]: dont hardcode names use {name: faker.random.string(20)}

File: test/models/AppModel/test.js [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+          },
+        },
+      ]);
+      [result] = aggResult.result;
+      expect(result).to.be.undefined;
+    });
+  });
+  describe('On updateOne', async () => {
+    let name, app, result;
+    beforeEach(async () => {
```
Comment (olegtsigulov) [inline test/models/AppModel/test.js]: make result const! you dont use it at beforeEach hook

File: test/models/AppModel/test.js [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+      name = faker.name.firstName();
+      app = await AppFactory.Create({ name });
+    });
+    it('Should update name at request and get true', async () => {
+      result = await AppModel.updateOne({ name, updData: { $set: { name: faker.name.firstName() } } });
+      expect(result.result).to.be.true;
```
Comment (olegtsigulov) [inline test/models/AppModel/test.js]: You can use destructurization (const {result} = ...) then in expect you dont need to result.result, just result

File: test/models/AppModel/test.js [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+    });
+    it('Should return false, when the data has not updated', async () => {
+      result = await AppModel.updateOne({ name: 'null', updData: { $set: { name: faker.name.firstName() } } });
+      expect(result.result).to.be.false;
```
Comment (olegtsigulov) [inline test/models/AppModel/test.js]: name: 'null' - hardcoded???

File: test/models/AppModel/test.js [hunk 18]
```diff
@@ -0,0 +1,131 @@
...
+    });
+    it('Should return duplicate key error, when the data has duplicate name', async () => {
+      const appDuplicate = await AppFactory.Create({ name: faker.name.firstName() });
+      result = await AppModel.updateOne({ name: app.name, updData: { $set: { name: appDuplicate.name } } });
+      expect(result.error).to.be.exist;
```
Comment (olegtsigulov) [inline test/models/AppModel/test.js]: use const {error} = ...

**Pull request merged**